Smarty/Smarty.class.php

```diff
@@ -1543,7 +1543,8 @@ function validateCompilePath( $compilePath ) {
         }
 
         $isValid = false;
-        if ( $fd = @fopen( $compilePath, 'wb') ) {
+        // Open file for writing but do not truncate.
+        if ( $fd = @fopen( $compilePath, 'c') ) {
             $isValid = true;
             @fclose( $fd );
             @unlink($compilePath);
```

Comment:
It looks like even if you remove these lines, templates are still recompiled, because $smarty->compile_check is